templates/workloads/partials/cf/service-base-properties.yml

```diff
@@ -40,12 +40,20 @@ NetworkConfiguration:
         - 0
         - Fn::Split:
           - ','
+          {{- if eq .Network.SubnetsType "IsolatedSubnets"}}
```

Comment:
I think we can skip this check completely!
https://github.com/aws/copilot-cli/blob/145cb1941917bad1fb7e6f1855652122636f4d86/internal/pkg/deploy/cloudformation/stack/transformers.go#L497-L500

If the "placement" is not "public" then we will automatically set this value to "PrivateSubnets"

Reply:
Ah. But the way that I handled distinguishing `private` and `isolated` for the envControllerParameters:
```
	if o.Network.SubnetsType == PrivateSubnetsPlacement {
		parameters = append(parameters, "NATWorkloads,") // YAML needs the comma separator; resolved in EnvContr.
	}
	if o.Network.SubnetsType == IsolatedSubnetsPlacement {
		parameters = append(parameters, "VPCEndpointWorkloads,")
	}
```
requires differentiation between the two types of "private" placements, so I did this:
```
if aws.StringValue(network.VPC.Placement) != manifest.PublicSubnetPlacement {
		opts.AssignPublicIP = template.DisablePublicIP
		switch aws.StringValue(network.VPC.Placement) {
		case manifest.PrivateSubnetPlacement:
			opts.SubnetsType = template.PrivateSubnetsPlacement
		case manifest.IsolatedSubnetPlacement:
			opts.SubnetsType = template.IsolatedSubnetsPlacement
		}
	}
```

Reply:
oh I see the concern, trying to think 💭  
My goal is to try to reduce the conditional logic in templates because it's just more difficult to test.

Instead maybe we can add a new field to `template.NetworkOpts` like this, what do you think? 
```go
type NetworkOpts struct {
   AssignPublicIP bool
   RequiresNAT    bool
   SubnetsType    string
   SecurityGroups []string
}
```